text/0000-panic-in-drop.md

```diff
@@ -0,0 +1,319 @@
+- Feature Name: `panic_in_drop`
+- Start Date: 2022-07-06
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Panics which happen inside a `Drop` impl currently unwind as long as the drop was not itself called during unwinding (which would otherwise abort immediated due to a double-panic). This RFC proposes to change this behavior to always abort if a panic attempts to escape a drop, even if this drop was invoked as part of normal function execution (as opposed to unwinding which already aborts with a double-panic).
+
+# Motivation
+[motivation]: #motivation
+
+There are two primary motivations for making this change: improving code size & compilation speed, and making it easier to write exception-safe unsafe code.
```

Comment:
I think in most discussions I've had the second of these is considered far more important and noteworthy - it would be useful I think to highlight that aspect here, since typically we don't focus on code size/compilation speed (and those are, I think, already achievable with -Cpanic=abort and -Zbuild-std, right?).